### PR TITLE
FEATURE: implement s3_canned_acl site setting

### PR DESCRIPTION
**Pull request by abelmokadem**
Implement S3 Canned ACL as a site setting

### CODE REVIEW

File: spec/components/file_store/s3_store_spec.rb [hunk 1]
```diff
@@ -250,6 +251,13 @@
 
   end
 
+  it "s3_canned_acl" do
```
Comment (ZogStriP) [inline spec/components/file_store/s3_store_spec.rb]: What's the point of this test? If it's testing the validation, then it should at least test for a bad case.

Also, the validation is already tested in `s3_canned_acl_site_setting_spec.rb`, so not sure what this is testing.

File: spec/models/s3_canned_acl_site_setting_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+describe S3CannedACLSiteSetting do
+
+  describe 'valid_value?' do
+    it 'returns true for a valid S3 region' do
```
Comment (ZogStriP) [inline spec/models/s3_canned_acl_site_setting_spec.rb]: I don't think that's the right description ;)

**Comment by SamSaffron** [comment]
on hold per: 

https://meta.discourse.org/t/migrating-uploads-to-s3-does-not-work-with-private-bucket/94591/6?u=sam

**Comment by SamSaffron** [comment]
So sorry closing this, lets figure out another way of doing this from a plugin without adding a site setting.

**Pull request closed without merging**